app/src/main/java/com/github/partymakers/partymaker/party/CreatePartyActivity.java

```diff
@@ -135,126 +232,147 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+            int currentDay = currentDate.get(Calendar.DAY_OF_MONTH);
 
             DatePickerDialog datePickerDialog = new DatePickerDialog(this, new DatePickerDialog.OnDateSetListener() {
                 @Override
                 public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
-                    viewBinding.textDatePicked.setText(dayOfMonth + "-" + ((monthOfYear + 1) <= 9 ? "0" + (monthOfYear + 1) : String.valueOf((monthOfYear + 1))) + "-" + year);
+                    Calendar date = Calendar.getInstance();
+                    if (party.getTimestamp() != null) {
+                        date.setTimeInMillis(party.getTimestamp());
+                    }
+                    date.set(year, monthOfYear, dayOfMonth);
+                    party.setTimestamp(date.getTimeInMillis());
+
+                    String dateText = SimpleDateFormat.getDateInstance(DateFormat.SHORT).format(date.getTime());
+                    viewBinding.textDatePicked.setText(dateText);
+//                    viewBinding.textDatePicked.setText(dayOfMonth + "-" + ((monthOfYear + 1) <= 9 ? "0" + (monthOfYear + 1) : String.valueOf((monthOfYear + 1))) + "-" + year);
                 }
-            }, year, month, day);
+            }, currentYear, currentMonth, currentDay);
             datePickerDialog.getDatePicker().setMinDate(System.currentTimeMillis()); //sets today's date as minimum date -> all the past dates are disabled
             datePickerDialog.show();
         } else if (v == viewBinding.textTimePicked) {
             // Get Current Time
             final Calendar calendar = Calendar.getInstance();
-            hour = calendar.get(Calendar.HOUR_OF_DAY);
-            minute = calendar.get(Calendar.MINUTE);
+            int hour = calendar.get(Calendar.HOUR_OF_DAY);
+            int minute = calendar.get(Calendar.MINUTE);
+
 
             // Launch Time Picker Dialog
-            TimePickerDialog timePickerDialog = new TimePickerDialog(this, new TimePickerDialog.OnTimeSetListener() {
+            new TimePickerDialog(this, new TimePickerDialog.OnTimeSetListener() {
                 @Override
-                public void onTimeSet(TimePicker view, int hourOfDay,
-                                      int minute) {
-                    viewBinding.textTimePicked.setText(hourOfDay + ":" + (minute <= 9 ? "0" + minute : String.valueOf(minute)));
+                public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
+                    Calendar time = Calendar.getInstance();
+                    if (party.getTimestamp() != null) {
+                        time.setTimeInMillis(party.getTimestamp());
+                    }
+                    time.set(Calendar.HOUR_OF_DAY, hourOfDay);
+                    time.set(Calendar.MINUTE, minute);
+                    party.setTimestamp(time.getTimeInMillis());
+                    String timeText = SimpleDateFormat.getTimeInstance(DateFormat.SHORT).format(time.getTime());
```

Comment:
Please use 'yyyy-mm-dd' or 'dd-mm-yyyyy' date formats  instead of 'mm-dd-yy'. 'mm-dd-yy' is common in the US only and might confuse users from other countries.